addons/io/org.openhab.io.transport.modbus/src/main/java/org/openhab/io/transport/modbus/internal/ModbusManagerImpl.java

```diff
@@ -0,0 +1,910 @@
+    }
+
+    @Override
+    public @Nullable EndpointPoolConfiguration getEndpointPoolConfiguration(ModbusSlaveEndpoint endpoint) {
+        Objects.requireNonNull(connectionFactory, "Not activated!");
+        return connectionFactory.getEndpointPoolConfiguration(endpoint);
+    }
+
+    @Override
+    public void addListener(ModbusManagerListener listener) {
+        listeners.add(listener);
+    }
+
+    @Override
+    public void removeListener(ModbusManagerListener listener) {
+        listeners.remove(listener);
+    }
+
+    @Override
+    public Set<@NonNull PollTask> getRegisteredRegularPolls() {
+        return this.scheduledPollTasks.keySet();
+    }
+
+    @Activate
+    protected void activate(Map<String, Object> configProperties) {
+        synchronized (this) {
+            logger.info("Modbus manager activated");
+            if (connectionPool == null) {
+                constructConnectionPool();
+            }
+            ScheduledExecutorService scheduledThreadPoolExecutor = this.scheduledThreadPoolExecutor;
+            if (scheduledThreadPoolExecutor == null) {
+                this.scheduledThreadPoolExecutor = scheduledThreadPoolExecutor = ThreadPoolManager
+                        .getScheduledPool(MODBUS_POLLER_THREAD_POOL_NAME);
+            }
+            ExecutorService callbackThreadPool = this.callbackThreadPool;
+            if (callbackThreadPool == null) {
+                this.callbackThreadPool = callbackThreadPool = ThreadPoolManager
+                        .getPool(MODBUS_POLLER_CALLBACK_THREAD_POOL_NAME);
+            }
+            if (scheduledThreadPoolExecutor.isShutdown() || callbackThreadPool.isShutdown()) {
+                logger.error("Thread pool(s) shut down! Aborting activation of ModbusMangerImpl");
+                throw new IllegalStateException("Thread pool(s) shut down! Aborting activation of ModbusMangerImpl");
+            }
+            monitorFuture = scheduledThreadPoolExecutor.scheduleWithFixedDelay(this::logTaskQueueInfo, 0,
+                    MONITOR_QUEUE_INTERVAL_MILLIS, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        synchronized (this) {
+            KeyedObjectPool<ModbusSlaveEndpoint, ModbusSlaveConnection> connectionPool = this.connectionPool;
+            if (connectionPool != null) {
+                Set<@NonNull PollTask> polls = getRegisteredRegularPolls();
+                for (PollTask task : polls) {
+                    unregisterRegularPoll(task);
+                }
+
+                connectionPool.close();
+                this.connectionPool = connectionPool = null;
+            }
+
+            if (monitorFuture != null) {
+                monitorFuture.cancel(true);
+                monitorFuture = null;
+            }
+            // Note that it is not allowed to shutdown the executor, since they will be reused when
+            // when pool is received from ExpressionThreadPoolManager is called
```

Comment:
ExpressionThreadPoolManager -> ThreadPoolManager

Reply:
Should be OK now